libs/langchain/langchain/document_loaders/parsers/pdf_docai.py

```diff
@@ -0,0 +1,217 @@
+"""Module contains a PDF parser based on DocAI from Google Cloud.
+
+You need to install two libraries to use this parser:
+pip install google-cloud-documentai
+pip install google-cloud-documentai-toolbox
+"""
+import time
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Iterator, List, Optional
+
+from google.api_core.operation import Operation
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseBlobParser
+from langchain.document_loaders.blob_loaders import Blob
+
+if TYPE_CHECKING:
+    from google.cloud.documentai import DocumentProcessorServiceClient
+
+
+@dataclass
+class DocAIParsingResults:
+    """A dataclass to store DocAI parsing results."""
+
+    source_path: str
+    parsed_path: str
+
+
+class DocAIPdfLoader(BaseBlobParser):
+    def __init__(
+        self,
+        project: str,
```

Comment:
Please add `*` operator prior to named arguments with defaults

```suggestion
        project: str,
        *,
```